jest/setup.js

```diff
@@ -1,15 +1,143 @@
+import fs from 'fs';
+import path from 'path';
 import 'react-native-gesture-handler/jestSetup';
+import _ from 'underscore';
 
 require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
 
 // Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
 jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
 
-// Set up manual mocks for methods used in the actions so our test does not fail.
-jest.mock('../src/libs/Notification/PushNotification', () => ({
-    // There is no need for a jest.fn() since we don't need to make assertions against it.
-    register: () => {},
-    deregister: () => {},
+jest.mock('react-native-blob-util', () => ({}));
+
+jest.mock('react-native-reanimated', () => {
+    const Reanimated = require('react-native-reanimated/mock');
+
+    // The mock for `call` immediately calls the callback which is incorrect
+    // So we override it with a no-op
+    Reanimated.default.call = () => {};
+    return Reanimated;
+});
+
+// <App> uses <ErrorBoundary> and we need to mock the imported crashlytics module
+// due to an error that happens otherwise https://github.com/invertase/react-native-firebase/issues/2475
+jest.mock('@react-native-firebase/crashlytics', () => () => ({
+    log: jest.fn(),
+    recordError: jest.fn(),
 }));
 
-jest.mock('react-native-blob-util', () => ({}));
+jest.mock('../src/libs/BootSplash', () => ({
+    hide: jest.fn(),
+    getVisibilityStatus: jest.fn().mockResolvedValue('hidden'),
+}));
+
+jest.mock('../src/libs/Notification/LocalNotification', () => ({
+    showCommentNotification: jest.fn(),
+}));
+
+/**
+ * @param {String} imagePath
+ */
+function mockImages(imagePath) {
+    const imageFilenames = fs.readdirSync(path.resolve(__dirname, `../assets/${imagePath}/`));
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    _.each(imageFilenames, (fileName) => {
+        if (/\.svg/.test(fileName)) {
+            jest.mock(`../assets/${imagePath}/${fileName}`, () => () => '');
+        }
+    });
+}
+
+// Mock all images so that Icons and other assets cannot break tests
+mockImages('images');
+mockImages('images/avatars');
+mockImages('images/bankicons');
+mockImages('images/product-illustrations');
+jest.mock('../src/components/Icon/Expensicons', () => ({
```

Comment:
Something that has never been clear to me... Why do we mock somethings in this file, but other things that get mocked live in the `__mock__` folder?

Is there a chance here for us to be consistent?

Reply:
Touched on in this follow-up issue: https://github.com/Expensify/Expensify/issues/228440 so we can discuss there and make this more standardized.